project/cmake/addons/CMakeLists.txt

```diff
@@ -197,10 +204,27 @@ foreach(addon ${addons})
         if(deflength GREATER 2 AND "${SOURCE_DIR}" STREQUAL "")
           list(GET def 2 revision)
 
+          # we need access to a git executable
+          find_package(Git REQUIRED)
+
+          # resolve revision to git hash
+          execute_process(COMMAND ${GIT_EXECUTABLE} ls-remote ${url} ${revision} OUTPUT_VARIABLE revision_hash)
+          # git ls-remote only works on branches and tag names but not on revisions
+          if(NOT "${revision_hash}" STREQUAL "")
+            string(REPLACE "\t" ";" revision_list ${revision_hash})
+            list(GET revision_list 0 revision_hash)
```

Comment:
Wait if we move this down below the `if(revision_hash AND NOT ${revision_hash} STREQUAL ${revision})` check it doesn't do the same anymore because after `execute_process()` we first need to sanitize `revision_hash` before we can compare it to `revision`.